subsys/net/ip/icmp.c

```diff
@@ -42,6 +42,10 @@ LOG_MODULE_REGISTER(net_icmp, MY_LOG_LEVEL);
 static K_MUTEX_DEFINE(lock);
 static sys_slist_t handlers;
 
+#if defined(CONFIG_NET_OFFLOAD)
```

Comment:
We should likely support socket offloading as well (` || defined(CONFIG_NET_SOCKETS_OFFLOAD)`), here and in other places.

Reply:
Sure, I had that in pipeline, the code was just not yet having that support.